### DISCUSSION TITLE
How to setup unique accessibility IDs for Capacitor components to use with Appium

### DISCUSSION THREAD

**Discussion by rocklaam** [appium/appium#17237] [Q&A]
Hey guys, my project has finished building an App using Capacitor and React. Currently, we are setting up a Java Appium project to automate test cases with this app. We are trying to add accessibility IDs or class names for React components like this:

![image](https://user-images.githubusercontent.com/29154789/179719235-f0eeef4e-928a-4190-aaa4-7337f03e5800.png)

However, when running the testing .apk (Android) and .ipa (iOS) file with the Appium inspector, all components were transformed into native flatform components like this:
**[Android]**
![image](https://user-images.githubusercontent.com/29154789/179719294-f5046e52-6038-43e6-a748-0bb2ec842616.png)

**and [iOS]**
![image](https://user-images.githubusercontent.com/29154789/179719467-f0538b3d-d7f2-431e-a159-ce5556438f04.png)

In Android, we can get component's ids via resource-id, but we can't in iOS builds.

Is our approach a proper solution or do we need to follow another way? In case our solution is correct, what do we need to get Ids for the iOS platform?

Looking forward to your answers, thank you very much!

**Comment by KazuCocoa** [comment]
Maybe https://discuss.appium.io/ is good place to ask such questions

**Comment by jlipps** [comment]
You might also reach out to the Capacitor folks to see if they have a way to set content description on Android, so you could set the same Accessibility ID across both platforms.

**Reply by rocklaam** [reply]
That's a good approach, thanks for your advice!

**Reply by facundofacio** [reply]
@rocklaam same issue here... did you have any luck with that?

**Comment by OleksiiRudyi** [comment]
Hi @rocklaam ! Did you find a solution to your problem? I'm facing the same issue. Though, I couldn't get the Accessibility ID on Android... For iOS this was the solution: the developer added the next code `aria-label={label}` and it was shown as Accessibility ID in Appium for iOS.
